### ISSUE TITLE
Strategies for limiting how fast Lambda pumps items into ElasticSearch?

### ISSUE THREAD

**Issue by MarkFischer** [awslabs/cloudwatch-logs-analyze-data#1]
I've been playing around with this model over the past few days, and am having troubles trying to analyze small data sets on a smallish Elasticsearch domain.  I'm seeing tons of errors like the following from my Lambda function:

EsRejectedExecutionException[rejected execution (queue capacity 50) ...

Reading up on this it seems like Lambda is just trying to send too much data to the Elasticsearch cluster at a time. There doesn't seem to be a way to increase the queue capacity that I can find, so all I can think of is attempting to limit how fast Lambda is pushing in events from the S3 bucket.

Is there any way to limit how many Lambda functions get spun up when getting put events from the S3 bucket? Or some other way to slow things down?

**Comment by Tammerabi** [comment]
Hi Mark, did you ever find a solution to this? I am running into the same situation.

**Comment by MarkFischer** [comment]
Nope.  I thought about looking into the lambda code itself, and introducing a manual delay, but never got around to doing it. Its a shame, without some way to verify that all the events are being ingested correctly I pretty much dropped this idea for now.

**Comment by smakintel** [comment]
using only REJECT type helps me to reduce the push rate
